src/sdk/namespace_fs.js

```diff
@@ -889,6 +891,23 @@ class NamespaceFS {
         }
     }
 
+    async _delete_object_empty_path(obj, fs_account_config) {
+        try {
+            let dir = path.dirname(obj);
+            while (dir !== this.bucket_path) {
+                const entries = await nb_native().fs.readdir(fs_account_config, dir);
```

Comment:
Why are you calling readdir?? This is very wasteful. Just try rmdir instead and catch errors - specifically ENOTEMPTY is expected, but also other errors can be ignored (after logging).